### PR TITLE
[WGSL] Update goals section

### PR DESCRIPTION
**Pull request by litherum**
... in accordance with
https://github.com/gpuweb/gpuweb/issues/586
and
https://github.com/gpuweb/gpuweb/issues/582

***
<a href="https://pr-preview.s3.amazonaws.com/litherum/gpuweb/pull/599.html" title="Last updated on Mar 10, 2020, 10:10 PM UTC (3c11631)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/gpuweb/gpuweb/599/e3b427d...litherum:3c11631.html" title="Last updated on Mar 10, 2020, 10:10 PM UTC (3c11631)">Diff</a>

### CODE REVIEW

File: wgsl/index.bs [hunk 1]
```diff
@@ -41,11 +41,32 @@ thead {
...
- * Trivially convertable to SPIR-V
- * Constructs are defined as normative references to their SPIR-V counterparts
- * All features in [SHORTNAME] are directly translatable to SPIR-V. (No polymorphism, no general pointers, no overloads, etc)
- * Features and semantics are exactly the ones of SPIR-V
- * Each item in this spec *must* provide the mapping to SPIR-V for the construct
+In no particular order:
+
+ * Text-based
+ * Readable and writable by programmers
+ * Mathematially [one-to-one](https://en.wikipedia.org/wiki/Injective_function) with SPIR-V
```
Comment (litherum) [inline wgsl/index.bs]: This is backwards! We want SPIR-V to be one-to-one with WGSL, not the other way around.

File: wgsl/index.bs [hunk 2]
```diff
@@ -41,11 +41,32 @@ thead {
...
+ * Supporting simple and efficient compilation to and from SPIR-V
+ * Supporting simple and efficient compilation to [Metal Shading Language](https://developer.apple.com/metal/Metal-Shading-Language-Specification.pdf) and [HLSL](https://docs.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl)
+ * Directly represents structured control flow
+ * Strongly statically typed
+ * As small as possible to represent what we need
```
Comment (litherum) [inline wgsl/index.bs]: Probably better to just to say "as small as possible." The other words aren't really necessary.

File: wgsl/index.bs [hunk 3]
```diff
@@ -41,11 +41,32 @@ thead {
 
 ## Goals ## {#goals}
 
- * Trivially convertable to SPIR-V
- * Constructs are defined as normative references to their SPIR-V counterparts
- * All features in [SHORTNAME] are directly translatable to SPIR-V. (No polymorphism, no general pointers, no overloads, etc)
- * Features and semantics are exactly the ones of SPIR-V
- * Each item in this spec *must* provide the mapping to SPIR-V for the construct
```
Comment (damyanp) [inline wgsl/index.bs]: > * Features and semantics are exactly the ones of SPIR-V
> * Each item in this spec *must* provide the mapping to SPIR-V for the construct

We seem to have lost these two goals, with in my opinion were pretty key to the entire WGSL proposal.

File: wgsl/index.bs [hunk 4]
```diff
@@ -41,11 +41,32 @@ thead {
...
+In no particular order:
+
+ * Text-based
```
Comment (damyanp) [inline wgsl/index.bs]: Text-based seeems redundant when there's also "readable and writable by programmers".  Can we imagine a WGSL spec that's readable and writable by programmers, but isn't also text based?

**Comment by dj2** [comment]
Closing for now, we can revisit after the FAQ work is done.

**Pull request closed without merging**